lib/engine/part/partition.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Part
+    class Partition < Base
+      attr_reader :a, :a_sign, :b, :b_sign, :type, :restrict, :magnet, :blockers, :inner, :outer
+
+      SIGN = {
+        '-' => -1,
+        nil => 0,
+        '+' => 1,
+      }.freeze
+
+      def initialize(a, b, type, restrict, magnet)
```

Comment:
what is magnet?

Reply:
It's just a rendering thing. I needed the river to go around this city:

![image](https://user-images.githubusercontent.com/3337865/101384583-87e1fa00-38b2-11eb-9511-6030d4a848ff.png)

and is also used here:

![image](https://user-images.githubusercontent.com/3337865/101384644-9b8d6080-38b2-11eb-85cd-fda01fb20dcb.png)